pkgs/by-name/sw/switch-to-configuration-ng/src/src/main.rs

```diff
@@ -931,6 +951,53 @@ fn remove_file_if_exists(p: impl AsRef<Path>) -> std::io::Result<()> {
     }
 }
 
+fn reexecute_systemd_manager(
+    dbus_conn: &LocalConnection,
+    fdo_dbus: &Proxy<'_, &LocalConnection>,
+) -> anyhow::Result<()> {
+    let reexecute_done = Rc::new(RefCell::new(false));
+    let _reexecute_done = reexecute_done.clone();
+    let owner_changed_token = fdo_dbus
+        .match_signal(
+            move |signal: OrgFreedesktopDBusNameOwnerChanged, _: &LocalConnection, _: &Message| {
+                if signal.name.as_str() == "org.freedesktop.systemd1" {
+                    *_reexecute_done.borrow_mut() = true;
+                }
+
+                true
+            },
+        )
+        .context("Failed to add signal match for DBus name owner changes")?;
+
+    let bus_name = BusName::from("org.freedesktop.systemd1");
+    let object_path = dbus::Path::from("/org/freedesktop/systemd1");
+    let interface = Interface::new("org.freedesktop.systemd1.Manager").unwrap();
+    let method_name = Member::new("Reexecute").unwrap();
```

Comment:
What are the failure modes here? Can we use `expect()` instead if we expect them to always succeed?

Reply:
Those functions return an error if the interface/member name does not follow the [rules](https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names) in the specification. But because they are constants here, we know that the names are valid and that no error will be returned.

I've changed to use an `expect()`. Tell me if you want me to change it back.